### PR TITLE
change specimen type and sample type to list

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

File: maestro-domain/src/main/java/bio/overture/maestro/domain/api/AnalysisCentricDocumentConverter.java [hunk 1]
```diff
@@ -80,19 +80,57 @@ private static AnalysisCentricDonor mergeDonorBySpecimen(@NonNull List<AnalysisC
     checkNotFound(anyDonor.getSpecimens() != null && anyDonor.getSpecimens().size() > 0,
             "Failed to merge AnalysisCentricDonor by specimen: donor doesn't have specimen,");
 
+    //  Each donor has only one specimen in the list
```
Comment (blabadi) [inline maestro-domain/src/main/java/bio/overture/maestro/domain/api/AnalysisCentricDocumentConverter.java:83]: how was this determined ?

**Comment by blabadi** [comment]
can you update at least one analysis in the song payloads fixtures to have multiple samples or/and specimens to cover the new logic, I don't think that is covered the way tests are now

**Comment by Buwujiu** [comment]
> can you update at least one analysis in the song payloads fixtures to have multiple samples or/and specimens to cover the new logic, I don't think that is covered the way tests are now

this test covers the logic where there are multiple specimen and multiple samples: `AnalysisCentricDocumentConverterTest` - `testGetDonors_multi_donor`

File: maestro-domain/src/main/java/bio/overture/maestro/domain/api/AnalysisCentricDocumentConverter.java (hunk 1 above)
Comment (Buwujiu) [reply maestro-domain/src/main/java/bio/overture/maestro/domain/api/AnalysisCentricDocumentConverter.java:83]: each song Sample has only one specimen and one donor,
so after groupingby by donor id , there is only one specimen

**Pull request merged**